src/main/scala/cognite/spark/v1/FlexibleDataModelCorePropertySyncRelation.scala

```diff
@@ -108,56 +137,84 @@ private[spark] class FlexibleDataModelCorePropertySyncRelation(
             properties = selectedInstanceProps.toIndexedSeq.filter(p =>
               !p.startsWith("node.") && !p.startsWith("edge.") && !p.startsWith("metadata.") &&
                 p != "startNode" && p != "endNode" && p != "space" && p != "externalId" && p != "type")
-        ))
+        )
+      )
       .toSeq
 
-    val instanceSyncRequest = InstanceSyncRequest(
+    InstanceSyncRequest(
       `with` = Map("sync" -> tableExpression),
-      cursors = if (cursor.nonEmpty) Some(Map("sync" -> cursor)) else None,
+      cursors = currentCursor.flatMap(c => Some(Map("sync" -> c))),
       select = Map("sync" -> SelectExpression(sources = sourceReference)),
       includeTyping = Some(true)
     )
+  }
 
-    Seq(syncOut(instanceSyncRequest, selectedColumns))
+  private def syncOut(
+      hasDataFilter: Boolean,
+      filters: Array[Filter],
+      selectedColumns: Array[String],
+      currentCursor: Option[String])(
+      implicit F: Async[IO]): Stream[IO, ProjectedFlexibleDataModelInstance] = {
+
+    val synced: IO[(ItemsWithCursor[InstanceDefinition], Boolean)] =
+      syncWithCursor(hasDataFilter, filters, selectedColumns, currentCursor)
+    val stream: Stream[IO, ProjectedFlexibleDataModelInstance] = Stream.eval(synced).flatMap {
+      case (ItemsWithCursor(items, nextCursor), continueWithHasData) =>
+        val projectedItems =
+          items.filter(clientsideHasDataFilter).map(toProjectedInstance(_, nextCursor, selectedColumns))
```

Comment:
just to double-check, we're still going to pass out the very last cursor via a column? What would be the way to pick up the most recent one in case of multiple page iterations? (safe to pick any but would be nice to pick latest, for that probably need some way to mark it, would passing a "page number" along with nextCursor be an option?)

Reply:
We pass cursor on every item, also along with node/edge.lastupdatedtime. If it is necessary to create logic to pick the latest, one could try to pick the one with highest lastupdatedtime. Not guaranteed to be latest cursor, but most likely is, or page -1.